orchagent/portsorch.cpp

```diff
@@ -1678,23 +1678,27 @@ bool PortsOrch::initPort(const string &alias, const set<int> &lane_set)
     return true;
 }
 
-void PortsOrch::deinitport(string alias, sai_object_id_t port_id)
+void PortsOrch::deinitport(const Port& p)
 {
     SWSS_LOG_ENTER();
 
-    Port p(alias, Port::PHY);
-    p.m_port_id = port_id;
+    string pAlias = p.m_alias;
+    sai_object_id_t pOID = p.m_port_id;
 
     /* remove port name map from counter table */
     RedisClient redisClient(m_counter_db.get());
-    redisClient.hdel(COUNTERS_PORT_NAME_MAP, alias);
+    redisClient.hdel(COUNTERS_PORT_NAME_MAP, pAlias);
+
+    // Destroy port's queue and PG flex stat counters
+    destroyQueueMapPerPort(p);
```

Comment:
should we have a per port flag to indicate whether the map was generated or not then decide to destroy it or not?

Reply:
Fair enough. That will help in the case where, flex counters are NOT enabled! will make the change.